modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryVersionLocalServiceImpl.java

```diff
@@ -39,11 +45,32 @@ public class ObjectEntryVersionLocalServiceImpl
 	public ObjectEntryVersion addObjectEntryVersion(ObjectEntry objectEntry)
 		throws PortalException {
 
-		return _updateObjectEntryVersion(
+		ObjectEntryVersion objectEntryVersion = _updateObjectEntryVersion(
 			objectEntry,
 			objectEntryVersionPersistence.create(
 				counterLocalService.increment()),
 			objectEntry.getVersion() + 1);
+
+		long objectEntryId = objectEntry.getObjectEntryId();
+
+		List<ObjectEntryVersion> versions = new ArrayList<>(
+			getObjectEntryVersions(objectEntryId));
+
+		if (_checkMaximumObjectEntryVersions(objectEntryId)) {
```

Comment:
I think the code would be more readable if we called this private method `_exceedsMaximumVersions(`